lib/src/main/java/com/dynatrace/file/util/DynatraceFileBasedConfigurationProvider.java

```diff
@@ -81,6 +85,7 @@ private void updateConfigFromFile(String fileName) {
         config.setMetricIngestToken(newToken);
       }
 
+      alreadyInitialized = true;
```

Comment:
No so sure about putting this here. :thinking: 
However, returning a boolean from this function to indicate success also does not feel right, and throwing an exception seems overkill. So I do not really have a better suggestion. :sweat_smile:

Reply:
I guess the idea was to use this variable to avoid logging that the token was refreshed on the first time it is read?

Reply:
I guess the idea was to use this variable to avoid logging that the token was refreshed on the first time it is read?

Reply:
Yes, exactly.

Reply:
Yes, I just thought it might be a bit detached from the actual initialization. I was expecting it to be set in `setUp()`. 
But it's just a nit anyway. Feel free to ignore it. :)

Reply:
Please update the PR title accordingly ;-)

Reply:
I left is as is now, as the whole thing is a bit "wonky", singletons should not actually have a set up method, afaik. I am not sure how to get around that in that case (especially when we want to keep this testable, but simultaneously make it super easy for users.